src/api/ApiClient.js

```diff
@@ -100,4 +128,33 @@ export default class ApiClient {
     }
     return res.messages;
   }
+
+  static async focusPing(account, hasFocus: boolean, newUserInput: boolean): Presences {
+    const res = await ApiClient.fetch(account, 'users/me/presence', {
+      method: 'post',
+      body: encodeAsURI({
+        status: hasFocus ? 'active' : 'idle',
+        new_user_input: newUserInput,
+      }),
+    });
+    if (res.result !== 'success') {
+      throw new Error(res.msg);
+    }
+    return res.presences;
+  }
+
+  static async messagesFlags(account, messages: number[], op: string, flag: string): number[] {
```

Comment:
messageIds sure is more clear, on the other hand the API call itself uses 'messages'...